code/modules/projectiles/projectile/energy/stun.dm

```diff
@@ -1,29 +1,345 @@
+	var/electrode_name
+
+/datum/status_effect/tased/on_creation(mob/living/new_owner, atom/fired_from, atom/firer, tase_stamina = 80, energy_drain = STANDARD_CELL_CHARGE * 0.05, electrode_name = "the electrodes")
+	if(isnull(fired_from) || isnull(firer) || !can_tase_with(fired_from))
+		qdel(src)
+		return
+
+	. = ..()
+	if(!.)
+		return
+
+	// does not use the status effect api because we snowflake it a bit
+	owner.throw_alert(type, /atom/movable/screen/alert/tazed)
+
+	set_taser(fired_from)
+	set_firer(firer)
+	src.stamina_per_second = tase_stamina
+	src.energy_drain = energy_drain
+	src.electrode_name = electrode_name
+
+/// Checks if the passed atom is captable of being used to tase someone
+/datum/status_effect/tased/proc/can_tase_with(atom/with_what)
+	if(istype(with_what, /obj/item/gun/energy))
+		var/obj/item/gun/energy/taser_gun = with_what
+		if(isnull(taser_gun.cell))
+			return FALSE
+
+	else if(istype(with_what, /obj/machinery))
+		var/obj/machinery/taser_machine = with_what
+		if(!taser_machine.is_operational)
+			return FALSE
+
+	return TRUE
+
+/// Actually does the tasing with the passed atom
+/// Returns TRUE if the tasing was successful, FALSE if it failed
+/datum/status_effect/tased/proc/do_tase_with(atom/with_what, seconds_between_ticks)
+	if(!can_see(taser, owner, 5))
+		return FALSE
+	if(istype(with_what, /obj/item/gun/energy))
+		var/obj/item/gun/energy/taser_gun = with_what
+		if(!taser_gun.cell?.use(energy_drain * seconds_between_ticks))
+			return FALSE
+		taser_gun.update_appearance()
+		return TRUE
+
+	if(istype(taser, /obj/machinery))
+		var/obj/machinery/taser_machine = taser
+		if(!taser_machine.is_operational)
+			return FALSE
+		if(!taser_machine.use_energy(energy_drain * seconds_between_ticks, force = FALSE))
+			return FALSE
+		return TRUE
+
+	if(istype(taser, /obj/item/mecha_parts/mecha_equipment))
+		var/obj/item/mecha_parts/mecha_equipment/taser_equipment = taser
+		if(!taser_equipment.chassis \
+			|| !taser_equipment.active \
+			|| taser_equipment.get_integrity() <= 1 \
+			|| taser_equipment.chassis.is_currently_ejecting \
+			|| taser_equipment.chassis.equipment_disabled \
+			|| !taser_equipment.chassis.use_energy(energy_drain * seconds_between_ticks))
+			return FALSE
+		return TRUE
+
+	return TRUE
+
+/datum/status_effect/tased/on_apply()
+	if(issilicon(owner) || isbot(owner) || isdrone(owner) || HAS_TRAIT(owner, TRAIT_PIERCEIMMUNE))
+		return FALSE
+
+	RegisterSignal(owner, COMSIG_LIVING_RESIST, PROC_REF(try_remove_taser))
+	RegisterSignal(owner, COMSIG_CARBON_PRE_MISC_HELP, PROC_REF(someome_removing_taser))
+	SEND_SIGNAL(owner, COMSIG_LIVING_MINOR_SHOCK)
+	owner.add_mood_event("tased", /datum/mood_event/tased)
+	owner.add_movespeed_modifier(/datum/movespeed_modifier/being_tased)
+	if(!HAS_TRAIT(owner, TRAIT_ANALGESIA))
+		owner.emote("scream")
+	if(HAS_TRAIT(owner, TRAIT_HULK))
+		owner.say(pick(
+			";RAAAAAAAARGH!",
+			";HNNNNNNNNNGGGGGGH!",
+			";GWAAAAAAAARRRHHH!",
+			"NNNNNNNNGGGGGGGGHH!",
```

Comment:
```suggestion
			";NNNNNNNNGGGGGGGGHH!",
```

Reply:
That is untouched from existing tasers so I'm keeping it as-is